## Final Project Submission

Please fill out:
* Student name: Ian Kiptoo
* Student pace: Full time Remote
* Scheduled project review date/time: 
* Instructor name: Asha Dean
* Blog post URL:


In [1]:
# Your code here - remember to use markdown cells for comments as well!

# 1.  Introduction
## 1.1 Overview
This project will entail using exploratory data analysis (EDA) to try and come up with useful insights that will be used by Microsoft to make informed bussiness decisions.
This is in regard to entering into a new market of origanal video production.

## 1.2 Business Understanding

Microsoft aims to gain a foothold in the film industry by launching its own film studio. However, lacking experience in film production, Microsoft sought to understand successful films to inform its strategy. By conducting exploratory data analysis (EDA), the project aims to identify key factors contributing to box office success, such as genre preferences, demographics and key responses. The insights gained will enable Microsoft to make informed decisions about the types of films to be released based on market trends and audience preferences. Ultimately, this move is intended to make Microsoft Movie Studios more competitive in the changing entertainment market.

## 1.3 Revieving my current working directory





In [2]:
ls

 Volume in drive C has no label.
 Volume Serial Number is 4298-8E60

 Directory of C:\Users\Ian\Documents\fraiton\dsc-data-science-env-config\dsc-phase-1-project-v2-4

03/18/2024  22:30    <DIR>          .
03/18/2024  22:30    <DIR>          ..
03/18/2024  19:27               163 .canvas
03/18/2024  19:27                84 .gitignore
03/18/2024  19:44    <DIR>          .ipynb_checkpoints
03/18/2024  19:27         1,425,341 awesome.gif
03/18/2024  19:27             1,846 CONTRIBUTING.md
03/18/2024  22:13                 0 im.db
03/18/2024  19:27             1,357 LICENSE.md
03/18/2024  19:27           107,563 movie_data_erd.jpeg
03/18/2024  19:27            22,781 README.md
03/18/2024  22:30            26,358 student.ipynb
03/18/2024  22:20    <DIR>          zippedData
               9 File(s)      1,585,493 bytes
               4 Dir(s)  47,049,347,072 bytes free


## 1.4 Importing all the relevant Libraries

In [3]:
import pandas as pd
import numpy as np
import sqlite3
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

# 2. Data Understanding and Data Analysis

## 2.1 Data understanding

### 2.1.1 Data sources

### 1.Box Office Mojo
This is an American website that provides box office data for movies. The site was founded in 1988 by Brandon Gray, and was bought in by IMDb in 2008. IMDb is itself ownwed by Amazon.
It specificy offers information about 

### 2. IMDB (Internet Movie Database)
IMDB is one of the most comprehensive database for films,television series, podcasts, home videos, video games, and streaming content online.This includes cast, production crew and personal biographies, plot summaries, trivia, ratings, and fan and critical reviews. It was founded in 1990, and moved to the web in 1993.

### 3. Rotten Tomatoes
Rotten Tomatoes is an American review aggregation website for film and television. It collects reviews from critics and audiences and assigns a "Tomatometer" score to movies based on the percentage of positive reviews. It also provides audience scores and reviews.

### 4. TheMovieDB
TheMovieDB (TMDB) is a user-generated database that provides information about movies, TV shows, and actors. It offers similar information to IMDB, including movie details, cast and crew information, user ratings, and more.

### 5. The Numbers
The numbers is a film industry data website that tracks box office revenue in a systematic, algorithimic way. 
The company also engagges in doing research and comes up income with forecasts of film projects.




### 2.1.2 Reading the data 

In [11]:
# Imported our various CSV's to understand contents as well

box_office_mojo_df = pd.read_csv('zippedData/bom.movie_gross.csv.gz')
rotten_tomatoes_df = pd.read_csv('zippedData/rt.movie_info.tsv.gz', delimiter = '\t')
themoviedb_df = pd.read_csv('zippedData/tmdb.movies.csv.gz')
the_numbers_df = pd.read_csv('zippedData/tn.movie_budgets.csv.gz')

### 2.1.3 Exploring the Data
Examination of the data structure and content. Understand the data types, dimensions, and any initial patterns or insights.

### 1. Box office Mojo

In [13]:
box_office_mojo_df.head()


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [14]:
box_office_mojo_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [15]:
# Calculate the total number of missing values in each column
missing_values_total = box_office_mojo_df.isnull().sum()

# Display the total number of missing values
print("Total missing values in each column:")
print(missing_values_total)

Total missing values in each column:
title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64


This data has a total of 5 columns and 3387 rows.
The columns names include title, studio, domestric_gross, foreign_gross and year.
In this data set we also have missing values. Three columns contains missing values, that is
studio with 5, domestric_gross with 28 and foreign_gross with 1350.

### Data types

### title 
This column contains object data type, likely representing titles of movies or similar categorical information.
### studio
Similarly, this column also contains object data type, likely representing the studios associated with the movies.
### domestic_gross
This column contains float64 data type, indicating numerical values for domestic gross earnings of movies.
### foreign_gross
This column contains object data type, which is unexpected for representing monetary values. It might need further investigation as it should ideally be of a numeric data type.
### year
This column contains int64 data type, representing the years associated with the movies.

Will need to do further analysis inorder to deal with the missing values and the foreign_gross column that is of type object which seems not to be what we expected of monetary terms.








### 2.Rotten Tomatoes

In [17]:
rotten_tomatoes_df.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [18]:
rotten_tomatoes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


In [19]:
# Calculate the total number of missing values in each column
missing_values_total = rotten_tomatoes_df.isnull().sum()

# Display the total number of missing values
print("Total missing values in each column:")
print(missing_values_total)

Total missing values in each column:
id                 0
synopsis          62
rating             3
genre              8
director         199
writer           449
theater_date     359
dvd_date         359
currency        1220
box_office      1220
runtime           30
studio          1066
dtype: int64


This data set contains 12 columns and 1560 rows.
The column names include id, synopsis, rating, genre, director, writer, theater_date, dvd_date, currency, box_office, runtime and studio.
In total we have 11 columns that contains missing values.
The only column that does not contain missing values is the id column.

The synopsis column has 62 missing values, the rating column has 3 missing values, the genre column has 8 missing values, the director column has 199 missing values, the writer column has 449 missing values, the theater_date column has 359 missing values, the dvd_date column has 359 missing values, the currency column has 1220 missing values, the box_office column has 1220 missing values, the runtime column has 30 missing values, and the studio column has 1066 missing values.

### Data types
### synopsis 
This column likely contains textual data and may be represented as an object data type.

### rating 
This column may represent categorical data such as movie ratings (e.g., PG, PG-13, R) and could be represented as an object data type.
### genre
This column may also represent categorical data indicating the genre of movies (e.g., Action, Comedy) and could be represented as an object data type.
### director
This column likely contains names of movie directors and may be represented as an object data type.
### writer
Similar to the director column, this column likely contains names of movie writers and may be represented as an object data type.
### theater_date and dvd_date
These columns may represent dates and could be represented as datetime data types.
### currency, box_office, and runtime
These columns may represent numerical data and could be represented as appropriate numerical data types (e.g., float64 for currency and box office, int64 for runtime).
### studio
This column may represent categorical data indicating the movie studios and could be represented as an object data type.

Gennerally the data types are what is expected, but on the other hand this data set contains a lot of missing values hence the need to do further analysis.

### 3. TheMovieDB

In [20]:
themoviedb_df.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [25]:
# removing the Unnamed column
themoviedb_df = pd.read_csv('zippedData/tmdb.movies.csv.gz', index_col=0)

In [26]:
themoviedb_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


In [27]:
# Calculate the total number of missing values in each column
missing_values_total = themoviedb_df.isnull().sum()

# Display the total number of missing values
print("Total missing values in each column:")
print(missing_values_total)

Total missing values in each column:
genre_ids            0
id                   0
original_language    0
original_title       0
popularity           0
release_date         0
title                0
vote_average         0
vote_count           0
dtype: int64


This data set contains 9 columns and 26517 rows.
The column names include genre_ids, id, original_language, originality_title, popularity, release_date, title, vote_average and vote_count.
This data set does not contain any missing values.

### Data types
### genre_ids
This column contains object data type, likely representing genre IDs associated with movies.
### id
This column contains int64 data type, representing unique identifiers for movies.
### original_language
This column contains object data type, indicating the original language of movies.
### original_title
This column contains object data type, likely representing the original titles of movies.
### popularity
This column contains float64 data type, representing popularity scores of movies.
### release_date
This column contains object data type, representing release dates of movies. It will be convinent to convert it to datetime data type for further analysis.
### title
This column contains object data type, representing the titles of movies.
### vote_average
This column contains float64 data type, representing average ratings of movies.
### vote_count
This column contains int64 data type, representing the number of votes cast for movies.


This data set does not have major issues and may be convient to use it.

### 4.The Numbers

In [33]:
the_numbers_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [34]:
the_numbers_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [31]:
# Calculate the total number of missing values in each column
missing_values_total = the_numbers_df.isnull().sum()

# Display the total number of missing values
print("Total missing values in each column:")
print(missing_values_total)

Total missing values in each column:
id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64


This data set contains 6 columns and 5782 rows.
The column names include id, release_date, movie, production_budget, domestic_gross and worlwide_gross.
Also this data does not contain any missing values.

### Data types
### id 
This column contains int64 data type, representing unique identifiers for movies.
### release_date
This column contains object data type, representing release dates of movies. It's convinent to convert it to datetime data type for further analysis.
### movie
This column contains object data type, likely representing the titles of movies.
### production_budget
This column contains object data type, which is unexpected for representing monetary values. It might need further investigation as it should ideally be of a numeric data type.
### domestic_gross 
This column contains object data type, which is unexpected for representing monetary 
### values
It might need further investigation as it should ideally be of a numeric data type.
### worldwide_gross
This column contains object data type, which is unexpected for representing monetary values. It might need further investigation as it should ideally be of a numeric data type.

This data requires further analysis inorder to determine the column types.

# 2.2 Data Understanding